lib/cuda/cudaDeviceInterface.cpp

```diff
@@ -15,6 +17,24 @@ cudaDeviceInterface::cudaDeviceInterface(Config& config_, int32_t gpu_id_, int g
     INFO("Number of CUDA GPUs: {:d}", max_num_gpus);
 
     cudaSetDevice(gpu_id);
+
+    // Find out how many GPUs clocks are allowed.
+    unsigned int mem_clock, core_clock, mem_count, core_count;
+    CHECK_CUDA_ERROR(nvmlDeviceGetSupportedMemoryClocks(gpu_id_, &mem_count,  &mem_clock));
+    CHECK_CUDA_ERROR(nvmlDeviceGetSupportedGraphicsClocks(gpu_id_, &core_count,  &core_clock));
+
+    INFO("Allowed GPU core clocks(MHz): ");
+    for (int i = 0; i < mem_count; ++i)  {
+        INFO("{:d}  ", mem_clock[i]);
+    }
+
+    INFO("Allowed GPU graphics clocks(MHz): ");
+    for (int i = 0; i < core_count; ++i)  {
+        INFO("{:d}  ", core_clock[i]);
+    }
+
+    CHECK_CUDA_ERROR(set_device_clocks(mem_clock, core_clock,
+                                       mem_count, core_count));
```

Comment:
```suggestion
    set_device_clocks(mem_clock, core_clock, mem_count, core_count));
```
The set clocks function is `void`, so it won't return anything for `CHECK_CUDA_ERROR` to check.